dspace-uclouvain/src/main/java/org/dspace/uclouvain/discovery/SolrServiceSessionYearFieldIndexingPlugin.java

```diff
@@ -22,6 +24,14 @@ public class SolrServiceSessionYearFieldIndexingPlugin implements SolrServiceInd
     @Autowired
     private ConfigurationService configurationService;
 
+    private MetadataField sessionField;
+    private MetadataField yearField;
+
+    public SolrServiceSessionYearFieldIndexingPlugin() throws Exception {
+        this.sessionField = new MetadataField(DSpaceServicesFactory.getInstance().getConfigurationService().getProperty("uclouvain.global.metadata.session.field", "masterthesis.session"));
+        this.yearField = new MetadataField(DSpaceServicesFactory.getInstance().getConfigurationService().getProperty("uclouvain.global.metadata.year.field", "dc.date.issued"));
+    }
```

Comment:
a little bit confused and not easy to read. Could be
```java
ConfigurationService configService = DSpaceServicesFactory.getInstance().getConfigurationService();
this.sessionField = new MetadataField(configService.getProperty("uclouvain.global.metadata.session.field", "masterthesis.session"));
this.yearField = new MetadataField(configService..getProperty("uclouvain.global.metadata.year.field", "dc.date.issued"));
```
or just store the `mdString` into class attribute and use `MetadataFieldService.find` to find the `MetadataField`. But this can't be done into constructor because `find` method needs `Context`